hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/TransactionManager.java

```diff
@@ -35,44 +35,43 @@
 public class TransactionManager implements Serializable {
 
   private static final Logger LOG = LogManager.getLogger(TransactionManager.class);
-
   private final LockManager lockManager;
+  private final boolean supportsOptimisticConcurrency;
```

Comment:
Name of the flag is misleading: had to go and check what it actually refers to to fully understand its semantic -- this one is rather about enabling/disabling CC (which you can disable if you only have a single writer)

Reply:
right, without changing the original config and all its usage, am going with `isOptimisticConcurrencyControlEnabled` flag in this class.